mdata/cache/cache_test.go

```diff
@@ -0,0 +1,118 @@
+package cache
+
+import (
+	"bytes"
+	"encoding/binary"
+	"testing"
+
+	"github.com/raintank/metrictank/iter"
+	"github.com/raintank/metrictank/mdata/chunk"
+)
+
+func getItgen(value uint32, ts uint32) *iter.IterGen {
+	var b []byte
+	buf := new(bytes.Buffer)
+	binary.Write(buf, binary.LittleEndian, uint8(chunk.FormatStandardGoTsz))
+	binary.Write(buf, binary.LittleEndian, uint32(value))
+	buf.Write(b)
+
+	itgen, _ := iter.NewGen(buf.Bytes(), ts)
+
+	return itgen
+}
+
+func getConnectedChunks(metric string) *CCache {
+	cc := NewCCache()
+
+	itgen1 := getItgen(1111111, 1000)
+	itgen2 := getItgen(1111111, 2000)
+	itgen3 := getItgen(1111111, 3000)
+	itgen4 := getItgen(1111111, 4000)
+	itgen5 := getItgen(1111111, 5000)
+
+	cc.Add(metric, 0, *itgen1)
+	cc.Add(metric, 1000, *itgen2)
+	cc.Add(metric, 2000, *itgen3)
+	cc.Add(metric, 3000, *itgen4)
+	cc.Add(metric, 4000, *itgen5)
+
+	return cc
+}
+
+func TestConsecutiveAdding(t *testing.T) {
```

Comment:
should we also test some other insertion scenarios? where the order is reversed or arbitrary, or gaps closed?